apps/web-client/src/lib/db/cr-sqlite/order-reconciliation.ts

```diff
@@ -210,6 +211,38 @@ export class ErrReconciliationOrderFinalized extends Error {
 		}
 	}
 }
+/**
+  * Deletes a reconciliation order and all its associated order lines from the
+ database.
+  * The deletion is performed as an atomic transaction to maintain data
+ consistency.
+  *
+  * @param db - The database connection
+  * @param id - The ID of the reconciliation order to delete
+  *
+  * @throws {Error} When:
+  * - The reconciliation order with the given ID is not found
+  * - The reconciliation order is already finalized
+  * - Database transaction fails
+  */
+export async function deleteReconciliationOrder(db: DB, id: number): Promise<void> {
+	const reconOrder = await db.execO<ReconciliationOrder>("SELECT * FROM reconciliation_order WHERE id = ?;", [id]);
+
+	if (!reconOrder[0]) {
+		throw new Error(`Reconciliation order ${id} not found`);
+	}
```

Comment:
Can we use `ErrReconciliationOrderNotFound` or `ErrReconciliationOrderFinalized` (depending on the case)...easier and more straight forward to test, provides for predictable expectations (when error handling) and already used in other places (already exists)